File: ch2/Linear_Search/algorithm.cs
using System;

public class algorithm 
{
    public static int linear_search(int[] L, int X) 
    {
        for(int i =0 ; i < L.Length ; i++) {
            if (L[i] == X) {
                return i;
            }
        }
        return -1;
    }

    public static void Main(string[] args) {
        int[] numbers = {1,4,123,123,54,3423};
        Console.Write(algorithm.linear_search(numbers, 54));
        Console.ReadKey();
    }

}

File: ch2/Linear_Search/algorithm.go
package main

import "fmt"

func linear_search(List []int, X int) int {
	for i:= 1; i < len(List); i++ {
		if List[i] == X {
			return i
		}
	}
	return -1
}

func main() {
	List := []int{5, 4, 1, 7, 4, 9, 10, 23, 12, 6, 3, 1}
	fmt.Println(linear_search(List, 23))
}